pkg/addon/addon.go

```diff
@@ -1092,6 +1085,23 @@ func checkDependencyNeedInstall(ctx context.Context, k8sClient client.Client, de
 		needInstallAddonDep = true
 		depClusters = addonClusters
 	} else {
+		// if addon clusters is nil, meaning to deploy to all clusters
+		if addonClusters == nil {
+			hasTopologyPolicy := false
+			topologyPolicyValue := map[string]interface{}{}
+			for _, policy := range depApp.Spec.Policies {
+				if policy.Type == "topology" {
+					hasTopologyPolicy = true
+					unmarshalErr := json.Unmarshal(policy.Properties.Raw, &topologyPolicyValue)
+					if unmarshalErr != nil {
+						return true, nil, unmarshalErr
+					}
+					break
+				}
+			}
+			need := hasTopologyPolicy && topologyPolicyValue["clusterLabelSelector"] == nil
+			return need, nil, nil
+		}
 		// get the runtime clusters of current dependency addon
 		for _, r := range depApp.Status.AppliedResources {
 			if r.Cluster != "" && !stringslices.Contains(depClusters, r.Cluster) {
```

Comment:
why we still need to judge the status of application?

Reply:
This is because the number of clusters may not be the same when between the fluxcd addon is installed and the rollout addon is installed.
When rollout clusters is nil, the clusters parameter of the fluxcd plugin is removed, so that the fluxcd addon is reinstalled on all clusters at the current time; However, if rollout clusters is not nil, the clusters parameter of the fluxcd addon will not be removed, and if the cluster is added between these two addon installations, the fluxcd plugin will not be re-installed. 
For example, when installing fluxcd for the first time without specifying clusters (`vela addon enable fluxcd`), there are c1 and c2 clusters. Then add c3 and install rollout(`vela addon enable rollout --clusters={c3}`). If use clusterLabelSelector to determine whether to install , then the fluxcd don't need to install when installing rollout, and we will miss the new joined c3 clusters for fluxcd.